### ISSUE TITLE
Swedish characters not being handled correctly

### ISSUE THREAD

**Issue by lfalck** [Adyen/adyen-mirakl#218]
Hi!

When we use the connector to send shops which have swedish letters (å, ä, ö) in for example the shareholder information they are not displayed correctly on the Adyen site.

When i call the Adyen API **updateAccountHolder** manually with Postman they are displayed correctly.

The characters are also displayed correctly when i fetch shops from Mirakl with the **api/shops** API.

Any ideas?

Shareholder address in Adyen portal:
![image](https://user-images.githubusercontent.com/30367604/69817551-21d08a80-11fb-11ea-8da7-4d4350b113a5.png)

**Comment by martinsrenato** [comment]
Hi @lfalck ,

Thanks for reaching out.
I ran some tests and these swedish characters seemed to work on my environment. But maybe you could try debugging the code and check if the characters are correct on the request that the connector sends to the Adyen API.
In the class [ShopService, method processUpdateAccountHolder](https://github.com/Adyen/adyen-mirakl/blob/aa32ef36df3754a0ecb50e05910285107e2b1249/src/main/java/com/adyen/mirakl/service/ShopService.java#L160), the object UpdateAccountHolderRequest contains the data that will be sent. You can also check the json that is generated during the `adyenAccountService.updateAccountHolder(updateAccountHolderRequest)` call.
If the characters are already incorrect before the call, it's possible that they are already coming like that from Mirakl.
Let me know what you find out.

Cheers,
Renato
Adyen

**Comment by lfalck** [comment]
Hi, thanks for helping out!

That's good that it works for you! Did you add the characters åäö to a shop in Mirakl or how did you test?

I have not debugged the code (not a Java developer :)) but i can see from the logs that the characters are wrong when fetching from Mirakl here:

```
Enter: com.adyen.mirakl.service.UboService.extractUbos() with argument[s] = [com.mirakl.client.mmp.domain.shop.MiraklShop@1774de, class GetAccountHolderResponse {
(...)
    accountHolderDetails: class AccountHolderDetails {
        metadata: {}
        address: class Address {
            city: Stockholm
            country: SE
            houseNumberOrName: 12
            postalCode: 11860
            stateOrProvince: null
            street: Ringv?gen 100
        }
```

I cannot see the updateAccountHolderRequest body in the logs but the UpdateAccountHolderResponse has the same invalid character as above.

```
Enter: com.adyen.mirakl.service.ShareholderMappingService.updateShareholderMapping() with argument[s] = [class UpdateAccountHolderResponse {
(...)
    accountHolderDetails: class AccountHolderDetails {
        metadata: {}
        address: class Address {
            city: Stockholm
            country: SE
            houseNumberOrName: 12
            postalCode: 11860
            stateOrProvince: null
            street: Ringv?gen 100
        }

```

**Comment by martinsrenato** [comment]
Hi, @lfalck!
Yes, I added åäö to a shop address, the connector retrieved the changes and sent to Adyen, and now it's displayed correctly when I check the shop on the portal.

The logs you mentioned are the data currently stored on Adyen side. Unfortunately the connector is not logging the requests by default. My point on checking the request was to understand if the "broken" characters are already set there, in that case I believe it would also indicate that the characters are already coming incorrectly from Mirakl.
For the request, you could log adding this line on the [processUpdateAccountHolder](https://github.com/Adyen/adyen-mirakl/blob/aa32ef36df3754a0ecb50e05910285107e2b1249/src/main/java/com/adyen/mirakl/service/ShopService.java#L160) method I mentioned before
       ` log.debug("updateAccountHolderRequest: {}", updateAccountHolderRequest);`

**Comment by lfalck** [comment]
I got some help from a colleague and we managed to solve the issue! 

This line was commented out in gradle.properties, when we enabled it the issue disappeared. I guess the releveant part is: `-Dfile.encoding=UTF-8 `

https://github.com/Adyen/adyen-mirakl/blob/aa32ef36df3754a0ecb50e05910285107e2b1249/gradle.properties#L58

I should perhaps mention that the connector is running on a Windows Server 2016 machine.
